### ISSUE TITLE
Follow Thread

### ISSUE THREAD

**Issue by silmaril42** [mastodon/mastodon#23702]
### Pitch

The possibility to follow a certain thread.

This means that all public and unlisted replies to the selected post will be shown in my home timeline, even if I don't follow the accounts posting them.

### Motivation

Consider this situation: you encounter an interesting post to which you don't really have to say something but you'd like to read further replies others might post later. Eg. a question for which you'd like to read the answers.

Currently one way to deal with this is keeping the thread open in a browser tab and reload it later, but that's nor really an elegant solution and it doesn't work across platforms / devices / apps.

Having this implemented as a core functionality would make this a lot easier and I think it might even facilitate new use cases for Mastodon.

**Comment by jimrhiz** [comment]
Bookmark it and read it again later from there? Bookmarks are good for remembering and returning to posts, and work on both the webui and apps.

**Comment by silmaril42** [comment]
Bookmarking is probably a better workaround than keeping tabs open - but it's still a workaround.

The main benefit of my suggestion is that new content in watched threads would come to my attention without any effort from my side.

This might be comparable to watching a thread in a web forum, which leads to e-mail notifications for new replies.

Thinking of this, following a thread could come with or without notifications (as following an account does now).

**Comment by trwnh** [comment]
this would require first-class support for the concept of a "thread" or "conversation" (not just for a "post"). currently, i think they are lazily created only when there are already multiple posts, not for single / one-off posts. you can't really do anything with them other than mute further notifications.

**Comment by raphmim** [comment]
+1 on this feature. 

Bookmarking definitely doesn't work, because you'd have to check on a regular basis to know whether there are new developments on the thread. 

But maybe following a post (not a thread) would be enough, so we don't need to wait for first-class support for thread?

**Comment by trwnh** [comment]
> But maybe following a post (not a thread) would be enough, so we don't need to wait for first-class support for thread?

That would be #7969

**Comment by kefir-** [comment]
Friendica already does this, even for discussions taking place on mastodon, so it should be doable.

https://friendica.myportal.social/display/e65e1095-1164-a326-7d70-ef2605053322

**Comment by kefir-** [comment]
From my comment on duplicate issue #25503:

If I'm reading this right, it looks like activitypub's InReplyTo would allow for the server side logic to resolve the chain of posts back to the thread start, and send notifications to anyone followup-subscribed as mentioned to one or more posts in that chain.

https://www.w3.org/TR/activitystreams-vocabulary/#dfn-inreplyto

**Comment by trwnh** [comment]
> activitypub's InReplyTo would allow for the server side logic to resolve the chain of posts back to the thread start

not necessarily. you may encounter an object in the chain that you can't resolve. also there isn't any limit on depth of recursion, which would lead to performance issues. more broadly, the construct of a "thread" is up to the receiving implementation.

**Comment by brendanjones** [comment]
I take it the difference between this and https://github.com/mastodon/mastodon/issues/7969 is that the request here is you get notified about any replies to a post at any depth of reply, whereas in #7969 you only get notified about top-level replies to a post? Have I understood that right?

**Comment by brendanjones** [comment]
Given I asked the same question as above on #7969 , and people answered number 4, that to me sounds like people expect to follow the thread ... in which case this issue is duplicate.

**Comment by kefir-** [comment]
Agree, this is a duplicate of #7969

**Comment by ldexterldesign** [comment]
@silmaril42 FYR terrible ambiguous uninformative title 🙂‍↔️
